plutus-ledger/src/Ledger/Tx/CardanoAPI.hs

```diff
@@ -718,6 +754,11 @@ toCardanoScriptInEra script = C.ScriptInEra C.PlutusScriptV1InAlonzo . C.PlutusS
 fromCardanoPlutusScript :: C.HasTypeProxy lang => C.PlutusScript lang -> P.Script
 fromCardanoPlutusScript = Codec.deserialise . BSL.fromStrict . C.serialiseToRawBytes
 
+fromCardanoPlutusScriptOrReferenceInput :: C.HasTypeProxy lang => C.PlutusScriptOrReferenceInput lang -> P.Script
+fromCardanoPlutusScriptOrReferenceInput = \case
+  C.PScript plutusScript -> fromCardanoPlutusScript plutusScript
+  C.PReferenceScript _   -> error ("TODO: implement PReferenceScript")
```

Comment:
@koslambrou what should we do in this case?

Reply:
Purge (or deprecate)? Ultimately, this shows a problem with the design of `TxIn` and `TxInType` in `plutus`. What's more, `fromCardanoTxInWitness` is not used anywhere else in the codebase, so I'm questionning it's use.

Another solution might be to move `TxInType` in `plutus-apps` and extend it to do something similar to `cardano-api`. I think `plutus` will probably remove it in the future as it's legacy code which isn't used by `cardano-ledger`. Am I right @michaelpj ?

Reply:
Um, I'm not quite following. What's wrong with `TxInType`? It is used as part of the `ScriptContext` so we can't change it super easily.

Reply:
Also, I can't seem to find where `TxInType` is used in the `ScriptContext`. Are you mixing it with `TxInInfo`?

Reply:
Yep, I'm 100% confused. Indeed, it seems those types should get purged from `plutus`!